### PR TITLE
[TACHYON-931] Improve the javadocs for the Tachyon Master

### PR DESCRIPTION
**Pull request by gpang**
https://tachyon.atlassian.net/browse/TACHYON-931

### CODE REVIEW

**Comment by calvinjia** [comment]
Jenkins, test this please.

File: servers/src/main/java/tachyon/master/MasterBase.java [hunk 1]
```diff
@@ -114,7 +117,9 @@ public void start(boolean isLeader) throws IOException {
       streamToJournalCheckpoint(checkpointStream);
       checkpointStream.close();
     } else {
-      // in standby mode. Start the journal tailer thread.
+      // This master is in standby mode. Start the journal tailer thread. Since the master is in
+      // standby mode, its RPC server is NOT serving. Therefore, the only thread modifying the
+      // master is the this journal tailer thread (no concurrent access).
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/MasterBase.java]: typo: remove "the"

File: servers/src/main/java/tachyon/master/journal/JournalOutputStream.java [hunk 2]
```diff
@@ -17,6 +17,10 @@
 
 import java.io.IOException;
 
+/**
+ * This output stream write {@link JournalEntry} objects and writes it to the journal. This output
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/journal/JournalOutputStream.java]: typo: write -> writes

File: servers/src/main/java/tachyon/master/MasterBase.java [hunk 3]
```diff
@@ -90,6 +90,9 @@ public void start(boolean isLeader) throws IOException {
        * Phase 3. Write out the checkpoint file. Since this master is completely up-to-date, it
        * writes out the checkpoint file. When the checkpoint file is closed, it will then delete the
        * complete log files.
+       *
+       * Since this method start() is called before the master RPC server starts serving, there is
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/MasterBase.java]: "this method" or "start()" but "this method start()" seems odd

File: servers/src/main/java/tachyon/master/block/meta/MasterBlockLocation.java [hunk 4]
```diff
@@ -29,15 +29,17 @@
...
 
+  /**
+   * @return the Tachyon worker id
+   */
   public long getWorkerId() {
     return mWorkerId;
   }
 
+  /**
+   * @return the tier that the block is on in this worker.
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/block/meta/MasterBlockLocation.java]: remove trailing period (or capitalize the first letter of the description)

File: servers/src/main/java/tachyon/master/journal/JournalOutputStream.java (hunk 2 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/journal/JournalOutputStream.java]: "write objects and writes it to ..." does not parse

File: servers/src/main/java/tachyon/master/journal/JournalReader.java [hunk 5]
```diff
@@ -29,7 +29,7 @@
 /**
  * This class manages reading from the journal. The reading must occur in two phases:
  *
- * 1. First the checkpoint file must be written.
+ * 1. First the checkpoint file must be read.
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/journal/JournalReader.java]: add a comma after "First"

File: servers/src/main/java/tachyon/master/journal/JournalReader.java [hunk 6]
```diff
@@ -72,6 +74,13 @@ public boolean isValid() {
     return mCheckpointRead && (mCheckpointOpenedTime == mCheckpointLastModifiedTime);
   }
 
+  /**
+   * Gets the {@link JournalInputStream} for the journal checkpoint file. This must be called before
+   * calling {@link #getNextInputStream()}.
+   *
+   * @return the {@link JournalInputStream} for the journal checkpoint file. This must be done
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/journal/JournalReader.java]: remove ". This must be done"

File: servers/src/main/java/tachyon/master/journal/JournalReader.java [hunk 7]
```diff
@@ -72,6 +74,13 @@ public boolean isValid() {
...
+   * @throws IOException
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/journal/JournalReader.java]: provide exception case description

**Comment by jsimsa** [comment]
Left you some comments.

File: servers/src/main/java/tachyon/master/MasterBase.java (hunk 1 above)
Comment (gpang) [inline servers/src/main/java/tachyon/master/MasterBase.java]: done

File: servers/src/main/java/tachyon/master/journal/JournalOutputStream.java (hunk 2 above)
Comment (gpang) [inline servers/src/main/java/tachyon/master/journal/JournalOutputStream.java]: fixed

File: servers/src/main/java/tachyon/master/MasterBase.java (hunk 3 above)
Comment (gpang) [inline servers/src/main/java/tachyon/master/MasterBase.java]: fixed

File: servers/src/main/java/tachyon/master/block/meta/MasterBlockLocation.java (hunk 4 above)
Comment (gpang) [inline servers/src/main/java/tachyon/master/block/meta/MasterBlockLocation.java]: removed

File: servers/src/main/java/tachyon/master/journal/JournalReader.java (hunk 5 above)
Comment (gpang) [inline servers/src/main/java/tachyon/master/journal/JournalReader.java]: added

File: servers/src/main/java/tachyon/master/journal/JournalReader.java (hunk 6 above)
Comment (gpang) [inline servers/src/main/java/tachyon/master/journal/JournalReader.java]: removed

File: servers/src/main/java/tachyon/master/journal/JournalReader.java (hunk 7 above)
Comment (gpang) [inline servers/src/main/java/tachyon/master/journal/JournalReader.java]: added

**Comment by gpang** [comment]
I fixed the documentation. Thanks for the reviews!

**Comment by jsimsa** [comment]
LGTM

**Comment by gpang** [comment]
I resolved the merge conflicts. Please take another look, @uronce-cc . Thanks.

**Comment by uronce-cc** [comment]
+1 LGTM

**Pull request merged**